### ISSUE TITLE
[BUG] Please authenticate via the warp client

### ISSUE THREAD

**Issue by jd-apprentice** [Boostport/setup-cloudflare-warp#109]
Hey! I have 3 similar workflows that run at the exact same hour and ran the same workflow, which is a notification to a gotify server that I'm running, after setting up everything for some reason? Sometimes it works, and other times shows the `Please authenticate via the warp client`

![Image](https://github.com/user-attachments/assets/c2775ac5-0d1a-4246-95e2-0401a2755aaa)

![Image](https://github.com/user-attachments/assets/5f78c6d2-18a7-4d27-bb52-ac4062c24d1c)

![Image](https://github.com/user-attachments/assets/62609ab8-5887-469c-af9f-9912f9918843)

I am doing something wrong?

**Comment by F21** [comment]
This seems to be some race condition. I know from personal experience, sometimes the warp client says it's connected when it isn't actually connected. As a test, can you try to add a sleep before the Gotify job and see if that makes it better?

**Comment by jd-apprentice** [comment]
![Image](https://github.com/user-attachments/assets/f2c54dda-0a89-4529-b059-870659de6e33)

I added a 10 seconds delay, but yet same result :(

**Comment by F21** [comment]
Are you able to show the commands for the Gotify job? Looking at the logs, it seems that Gotify is having trouble detecting WARP as connected, as the "Please authenticate via the warp client" message does not seem to be from the warp-cli.

**Comment by jd-apprentice** [comment]
Yeah! you can take a look at my actions [here](https://github.com/jd-apprentice/jd-server/tree/main/.github/workflows) the one running warp-cli is called `cloudflare-auth`

**Comment by F21** [comment]
Your command seems to be a pretty straight forward curl call to your server, so I am not really sure why this is happening. I know I sometimes get [failures](https://github.com/Boostport/setup-cloudflare-warp/actions/workflows/pull-requests.yml?query=is%3Afailure) as well when running tests for pull requests for this action, but they pass when I retry them, so the problem seems to be intermittent. 

Can you update your script to retrieve the warp debug logs, upload them as a GitHub artifact and see if you can find anything useful in them?
See:
- https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/troubleshooting/warp-logs/
- https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/storing-and-sharing-data-from-a-workflow#uploading-build-and-test-artifacts
- https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/downloading-workflow-artifacts

**Comment by jd-apprentice** [comment]
Okie! In the week I'll give it a shot and see whats going on :)

**Comment by Krymonota** [comment]
I've encountered the same issue. The response body `Please authenticate via the warp client` does indeed come from Cloudflare. I realized this is expected behavior because the response is a 302 redirect to `https://ORGANIZATION.cloudflareaccess.com/cdn-cgi/access/login/...`, and curl doesn't follow redirects by default. It works perfectly when passing the `-L` parameter to curl.
